### PR TITLE
fix for case where nothing is returned from vswhere

### PR DESCRIPTION
**Pull request by dmoody256**
Small change to protect against the case that no msvc is installed.

## Contributor Checklist:

* [x] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [ ] I have updated `master/src/CHANGES.txt` directory (and read the `README.txt` in that directory)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

File: src/engine/SCons/Tool/MSCommon/vc.py [hunk 1]
```diff
@@ -298,13 +298,14 @@ def find_vc_pdir_vswhere(msvc_version):
...
         sp = subprocess.Popen(vswhere_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
         vsdir, err = sp.communicate()
-        vsdir = vsdir.decode("mbcs").splitlines()
-        # vswhere could easily return multiple lines
-        # we could define a way to pick the one we prefer, but since
-        # this data is currently only used to make a check for existence,
-        # returning the first hit should be good enough for now.
-        vc_pdir = os.path.join(vsdir[0], 'VC')
-        return vc_pdir
+        if vsdir:
```
Comment (bdbaddog) [inline src/engine/SCons/Tool/MSCommon/vc.py:301]: Is there an error code which we're ignoring and could be useful here?

**Comment by bdbaddog** [comment]
Nice mocked test!

**Comment by bdbaddog** [comment]
Can you change the test to use a fixture for the files? 
https://github.com/SCons/scons/wiki/TestingMethodology#working-with-fixtures

File: test/MSVC/no_msvc.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+# test find_vc_pdir_vswhere by removing all other VS's reg keys
+test.write('SConstruct', """
+import SCons.Tool.MSCommon
+
+for key in SCons.Tool.MSCommon.vc._VCVER_TO_PRODUCT_DIR:
+    SCons.Tool.MSCommon.vc._VCVER_TO_PRODUCT_DIR[key]=[(SCons.Util.HKEY_LOCAL_MACHINE, r'')]
+
+env = Environment()
+""" % locals())
```
Comment (bdbaddog) [inline test/MSVC/no_msvc.py]: No need for locals()?

File: test/MSVC/no_msvc.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    return None
+
+for key in SCons.Tool.MSCommon.vc._VCVER_TO_PRODUCT_DIR:
+    SCons.Tool.MSCommon.vc._VCVER_TO_PRODUCT_DIR[key]=[(SCons.Util.HKEY_LOCAL_MACHINE, r'')]
+
+SCons.Tool.MSCommon.vc.find_vc_pdir_vswhere = DummyVsWhere
+
+env = Environment()
+print('MSVC_VERSION='+str(env.get('MSVC_VERSION')))
+""" % locals())
```
Comment (bdbaddog) [inline test/MSVC/no_msvc.py]: no need for locals()?

File: src/engine/SCons/Tool/MSCommon/vc.py (hunk 1 above)
Comment (dmoody256) [reply src/engine/SCons/Tool/MSCommon/vc.py:301]: seems that it returns success even when it finds no installs, so empty output is the way to check. See for more info: https://github.com/Microsoft/vswhere/issues/65

File: test/MSVC/no_msvc.py (hunk 2 above)
Comment (dmoody256) [reply test/MSVC/no_msvc.py]: used a fixture in 68069af

File: test/MSVC/no_msvc.py (hunk 3 above)
Comment (dmoody256) [reply test/MSVC/no_msvc.py]: used a fixture in 68069af

**Comment by bdbaddog** [comment]
Update CHANGES.txt and we should be good to go?

File: src/CHANGES.txt [hunk 4]
```diff
@@ -18,6 +18,7 @@ RELEASE  VERSION/DATE TO BE FILLED IN LATER
   From Daniel Moody:
     - Change the default for AppendENVPath to delete_existing=0, so path
       order will not be changed, unless explicitly set (Issue #3276)
+    - fix bug where no msvc's are found and an empty list is indexed (PR# #3288)
```
Comment (bdbaddog) [inline src/CHANGES.txt]: `Fix bug causing error on windows when no MSVC installation is available.  `

No need to self reference the PR. Only reference an Issue if one's been filed.

Sorry to be a bit nitpicky about this but CHANGES.txt is where people often look to see if their problem has been fixed.

Comment (dmoody256) [reply src/CHANGES.txt]: remove PR reference in 6a61fae

**Pull request merged**